tests/test_chat_handler.py

```
@@ -0,0 +1,457 @@
+        json.loads(line) for line in log.read_text(encoding="utf-8").splitlines() if line.strip()
+    ]
+    cap_hits = [evt for evt in lines if evt.get("event_type") == "chat_cap_hit"]
+    assert len(cap_hits) == 1
+    assert cap_hits[0]["cap_kind"] == "input"
+
+
+# ── end-to-end run_turn (mocked LLM) ───────────────────────────
+
+
+@pytest.fixture
+def _stub_litellm(monkeypatch):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_stub_litellm`

(ANN202)